sklearn/neighbors/_quad_tree.pyx

```diff
@@ -39,18 +41,18 @@ CELL_DTYPE = np.dtype({
                 np.intp, np.intp, (np.float32, 3), (np.float32, 3),
                 (np.float32, 3), (np.float32, 3)],
     'offsets': [
-        <Py_ssize_t> &(<Cell*> NULL).parent,
-        <Py_ssize_t> &(<Cell*> NULL).children,
-        <Py_ssize_t> &(<Cell*> NULL).cell_id,
-        <Py_ssize_t> &(<Cell*> NULL).point_index,
-        <Py_ssize_t> &(<Cell*> NULL).is_leaf,
-        <Py_ssize_t> &(<Cell*> NULL).squared_max_width,
-        <Py_ssize_t> &(<Cell*> NULL).depth,
-        <Py_ssize_t> &(<Cell*> NULL).cumulative_size,
-        <Py_ssize_t> &(<Cell*> NULL).center,
-        <Py_ssize_t> &(<Cell*> NULL).barycenter,
-        <Py_ssize_t> &(<Cell*> NULL).min_bounds,
-        <Py_ssize_t> &(<Cell*> NULL).max_bounds,
+        (<Py_ssize_t>&(dummy.parent) - <Py_ssize_t>&(dummy)),
```

Comment:
The mailing list states to do this:

```
cdef Struct tmp
cdef Py_ssize_t offset
offset = <Py_ssize_t> (<Py_intptr_t>&(tmp.field) - <Py_intptr_t>&(tmp))
```

with castinig to `Py_intptr_t` first, do you think this is better?

Reply:
Hi! Thanks for reviewing this.

I think some form of `intptr_t` is the right tool for the job, and I tried `Py_intptr_t` first, but I couldn't make it work because of `'Py_intptr_t' is not a type identifier` errors when I tried to build.

I haven't been able to find `Py_intptr_t` available as an import anywhere, but I've digged a little more and I've found that we can import the actual C `intptr_t` type from `libc.stdint` and it works.

Should I go with that instead? Do you have any advice about where to get `Py_intptr_t` from?

Reply:
I think the current implementation is okay. It could be that `Py_ssize_t` is typedefed to `Py_intptr_t` anyways:

https://github.com/cython/cython/blob/0a25de5b1023fd551f5e600583de8d8c87b7796d/Cython/Utility/MemoryView.pyx#L37

Reply:
Sorry for the github newbie question: if you think the current implementation is okay should I mark this conversation as resolved?

Reply:
I personally do not resolve them on my PRs, because resolved conversations are automatically hidden and could be interesting to other reviewers.

That being said, you can mark the conversation as resolved.

Reply:
Thanks for the explanation. I'll leave it open then :)